File: University-Work/Java/App-Development_and_Mobility/Evenemang App/app/src/main/java/com/example/lab3/MainActivity.java
package com.example.lab3;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

/**
 * The MainActivity class is the entry point of the application that
 * displays the login button to the user and redirects to the EventsActivity.
 *
 * @author Adam El Soudi
 * @version 1.0
 * @since 14/03/23
 */

public class MainActivity extends AppCompatActivity {

    MyDatabaseHandler db = new MyDatabaseHandler(this);
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button myButton = findViewById(R.id.login_button);
        myButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, EventsActivity.class);
                startActivity(intent);

                //db.deletePost(7);

            }
        });
    }
}

File: University-Work/Kotlin/SpaceShooter/app/src/main/java/com/example/spaceshooter/MainActivity.kt
package com.example.spaceshooter

import android.annotation.SuppressLint
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView

private const val TAG = "MainActivity"
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d(TAG, "Deleted db")
        val dbHelper = DatabaseHelper(this)
        //dbHelper.deleteDatabase()

        findViewById<Button>(R.id.startGameButton)?.setOnClickListener {
            val intent = Intent(this, GameActivity::class.java)
            startActivity(intent)
        }

        findViewById<Button>(R.id.highScoresButton)?.setOnClickListener {
            val backIntent = Intent(this, HighScoresActivity::class.java)
            startActivity(backIntent)
        }
    }


    override fun onResume() {
        super.onResume()

        //val prefs = getSharedPreferences(PREFS, MODE_PRIVATE)
        //val longestDistance = prefs.getInt(LONGEST_DIST, 0)
        val highscoreTextView = findViewById<TextView>(R.id.highscore)
        val highScoreManager = HighScoreManager(this)
        val scores = highScoreManager.getScores()

        if (scores.isNotEmpty()) {
            val topScore = scores[0]
            highscoreTextView.text = getString(R.string.longest_distance, topScore)
        } else {
            highscoreTextView.text = getString(R.string.no_scores_available)
        }
    }
}